File: app/src/main/java/com/toshi/util/ResendHandler.java
package com.toshi.util;

import android.support.design.widget.BottomSheetDialog;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.LinearLayout;

import com.toshi.R;
import com.toshi.model.local.User;
import com.toshi.model.sofa.Payment;
import com.toshi.model.sofa.PaymentRequest;
import com.toshi.view.adapter.listeners.OnUpdateListener;
import com.toshi.view.fragment.DialogFragment.PaymentConfirmationDialog;

public class ResendHandler {

    private AppCompatActivity activity;

    public ResendHandler(final AppCompatActivity activity) {
        this.activity = activity;
    }

    public void showResendPaymentConfirmationDialog(final User receiver,
                                                    final Payment payment,
                                                    final OnUpdateListener listener) {
        final PaymentConfirmationDialog dialog =
                PaymentConfirmationDialog.newInstanceToshiPayment(
                        receiver.getToshiId(),
                        payment.getValue(),
                        null,
                        PaymentType.TYPE_SEND
                );

        dialog.show(this.activity.getSupportFragmentManager(), PaymentConfirmationDialog.TAG);
        dialog.setOnPaymentConfirmationApprovedListener(__ -> listener.onUpdate());
    }

    public void showPaymentConfirmationDialog(final User receiver,
                                              final String amount,
                                              final OnUpdateListener listener) {
        final PaymentConfirmationDialog dialog =
                PaymentConfirmationDialog.newInstanceToshiPayment(
                        receiver.getToshiId(),
                        amount,
                        null,
                        PaymentType.TYPE_SEND
                );
        dialog.show(this.activity.getSupportFragmentManager(), PaymentConfirmationDialog.TAG);
        dialog.setOnPaymentConfirmationApprovedListener(__ ->
                listener.onUpdate()
        );
    }

    public void showPaymentRequestConfirmationDialog(final User receiver,
                                                     final PaymentRequest paymentRequest,
                                                     final OnUpdateListener listener) {
        final PaymentConfirmationDialog dialog =
                PaymentConfirmationDialog.newInstanceToshiPayment(
                        receiver.getToshiId(),
                        paymentRequest.getValue(),
                        null,
                        PaymentType.TYPE_REQUEST
                );
        dialog.show(this.activity.getSupportFragmentManager(), PaymentConfirmationDialog.TAG);
        dialog.setOnPaymentConfirmationApprovedListener(__ -> {
            listener.onUpdate();
        });
    }

    public void showResendDialog(final OnUpdateListener listener,
                                 final OnUpdateListener deleteListener) {
        final BottomSheetDialog resendDialog = new BottomSheetDialog(this.activity);
        final View sheetView = this.activity.getLayoutInflater().inflate(R.layout.view_chat_resend, null);
        final LinearLayout deleteView = sheetView.findViewById(R.id.deleteMessage);
        deleteView.setOnClickListener(view -> {
            resendDialog.dismiss();
            deleteListener.onUpdate();
        });
        final LinearLayout retryView = sheetView.findViewById(R.id.retry);
        retryView.setOnClickListener(__ -> {
            resendDialog.dismiss();
            listener.onUpdate();
        });
        resendDialog.setContentView(sheetView);
        resendDialog.show();
    }
}

File: app/src/main/java/com/toshi/util/ResendHandler.kt
package com.toshi.util

import android.support.design.widget.BottomSheetDialog
import android.support.v7.app.AppCompatActivity
import android.widget.LinearLayout
import com.toshi.R
import com.toshi.manager.model.PaymentTask
import com.toshi.model.local.User
import com.toshi.model.sofa.Payment
import com.toshi.model.sofa.PaymentRequest
import com.toshi.view.fragment.PaymentConfirmationFragment

class ResendHandler(private val activity: AppCompatActivity) {

    fun showResendPaymentConfirmationDialog(receiver: User,
                                            payment: Payment,
                                            listener: (PaymentTask) -> Unit) {
        val dialog = PaymentConfirmationFragment.newInstanceToshiPayment(
                receiver.toshiId,
                payment.value, null,
                PaymentType.TYPE_SEND
        )

        dialog.show(this.activity.supportFragmentManager, PaymentConfirmationFragment.TAG)
        dialog.setOnPaymentConfirmationApprovedListener { listener(it) }
    }

    fun showPaymentConfirmationDialog(receiver: User,
                                      amount: String,
                                      listener: (PaymentTask) -> Unit) {
        val dialog = PaymentConfirmationFragment.newInstanceToshiPayment(
                receiver.toshiId,
                amount, null,
                PaymentType.TYPE_SEND
        )
        dialog.show(this.activity.supportFragmentManager, PaymentConfirmationFragment.TAG)
        dialog.setOnPaymentConfirmationApprovedListener { listener(it) }
    }

    fun showPaymentRequestConfirmationDialog(receiver: User,
                                             paymentRequest: PaymentRequest,
                                             listener: (PaymentTask) -> Unit) {
        val dialog = PaymentConfirmationFragment.newInstanceToshiPayment(
                receiver.toshiId,
                paymentRequest.value, null,
                PaymentType.TYPE_REQUEST
        )
        dialog.show(this.activity.supportFragmentManager, PaymentConfirmationFragment.TAG)
        dialog.setOnPaymentConfirmationApprovedListener { listener(it) }
    }

    fun showResendDialog(listener: () -> Unit,
                         deleteListener: () -> Unit) {
        val resendDialog = BottomSheetDialog(this.activity)
        val sheetView = this.activity.layoutInflater.inflate(R.layout.view_chat_resend, null)
        val deleteView = sheetView.findViewById<LinearLayout>(R.id.deleteMessage)
        deleteView.setOnClickListener {
            resendDialog.dismiss()
            deleteListener()
        }
        val retryView = sheetView.findViewById<LinearLayout>(R.id.retry)
        retryView.setOnClickListener {
            resendDialog.dismiss()
            listener()
        }
        resendDialog.setContentView(sheetView)
        resendDialog.show()
    }
}
